### PR TITLE
Add net.openra.OpenRA

### PR DESCRIPTION
**Pull request by Unrud**
The Lua patches are from https://git.archlinux.org/svntogit/packages.git/tree/trunk?h=packages/lua51

### CODE REVIEW

File: net.openra.OpenRA.json [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    "--share=network",
+    "--persist=.openra"
+  ],
+  "modules": [
+    {
+      "name": "mono",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.mono-project.com/sources/mono/mono-5.4.1.6.tar.bz2",
```
Comment (TingPing) [inline net.openra.OpenRA.json]: Outdated version.

File: net.openra.OpenRA.json [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+  "finish-args": [
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--device=dri",
+    "--share=network",
+    "--persist=.openra"
+  ],
+  "modules": [
+    {
+      "name": "mono",
```
Comment (TingPing) [inline net.openra.OpenRA.json]: There is a mono sdk packaged: https://github.com/flathub/org.freedesktop.Sdk.Extension.mono5

File: net.openra.OpenRA.json [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        "make prefix=/app install-linux-appdata"
+      ]
+    },
+    {
+      "name": "export-other-mods",
+      "buildsystem": "simple",
+      "build-commands": [
+        "sed -i 's/<name>OpenRA - Red Alert/<name>OpenRA/' /app/share/appdata/openra-ra.appdata.xml",
+        "sed -i 's/<id>openra/<id>net.openra.OpenRA.openra/' /app/share/appdata/openra-{cnc,d2k}.appdata.xml",
+        "sed -i 's/Icon=openra/Icon=net.openra.OpenRA.openra/' /app/share/applications/openra-{cnc,d2k}.desktop",
```
Comment (Barthalion) [inline net.openra.OpenRA.json]: I think it's being fixed automatically (but I admit that I can be wrong about it).

File: net.openra.OpenRA.json [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+          "commands": [
+            "sed -e 's:llua:llua5.1:' -e 's:/include:/include/lua5.1:' -i etc/lua.pc",
+            "sed -r -e '/^LUA_(SO|A|T)=/ s/lua/lua5.1/' -e '/^LUAC_T=/ s/luac/luac5.1/' -i src/Makefile",
+            "mv doc/lua.1 doc/lua5.1.1",
+            "mv doc/luac.1 doc/luac5.1.1"
+          ]
+        }
+      ],
+      "cleanup": [ "/bin", "/include", "/lib/*.a", "/lib/*.la", "/lib/pkgconfig", "/share/man" ],
+      "buildsystem": "simple",
```
Comment (Barthalion) [inline net.openra.OpenRA.json]: Any idea if it works with luajit?

File: net.openra.OpenRA.json (hunk 3 above)
Comment (Unrud) [reply net.openra.OpenRA.json]: It's not fixed automatically.

File: net.openra.OpenRA.json (hunk 4 above)
Comment (Unrud) [reply net.openra.OpenRA.json]: OpenRA uses **MemoryConstrainedLuaRuntime** from [Eluant](https://github.com/cdhowie/Eluant). Unfortunately is uses a custom memory allocator, which doesn't meet the special needs of LuaJIT on 64 bit.

File: net.openra.OpenRA.json (hunk 2 above)
Comment (Unrud) [reply net.openra.OpenRA.json]: Fixed

**Comment by Barthalion** [comment]
LGTM. We're discussing making [OARS](https://odrs.gnome.org/oars) mandatory for new applications, so while it's not required yet, it would be nice to have it in the appdata.

bot, build net.openra.OpenRA

**Comment by nedrichards** [comment]
Yeah, I'd love to see OARS here.

**Comment by Unrud** [comment]
I think that the appdata already contains OARS.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/net.openra.OpenRA

**Comment by nedrichards** [comment]
@Unrud I've sent you an invitation to join the new project so you can push improvements and updates. If there's anyone else who should be added, let us know.

**Pull request closed without merging**